src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Pipeline.cs

```diff
@@ -0,0 +1,498 @@
+        public PipelineState GetState()
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            PipelineState retState = PipelineState.Unknown;
+            NNStreamerError ret = NNStreamerError.None;
+
+            ret = Interop.Pipeline.GetState(_handle, out retState);
+            NNStreamer.CheckException(ret, "Failed to get the pipeline state because of internal error");
+
+            return retState;
+        }
+
+        /// <summary>
+        /// Starts the pipeline, asynchronously. (The state would be changed to PipelineState.Playing)
+        /// </summary>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to start the pipeline.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Start()
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            NNStreamerError ret = NNStreamerError.None;
+
+            ret = Interop.Pipeline.Start(_handle);
+            NNStreamer.CheckException(ret, "Failed to start the pipeline because of internal error");
+        }
+
+        /// <summary>
+        /// Stops the pipeline, asynchronously. (The state would be changed to PipelineState.Paused)
+        /// </summary>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to stop the pipeline.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Stop()
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            NNStreamerError ret = NNStreamerError.None;
+
+            ret = Interop.Pipeline.Stop(_handle);
+            NNStreamer.CheckException(ret, "Failed to stop the pipeline because of internal error");
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object including opened handle.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        /// <since_tizen> 8 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                // release managed object
+            }
+
+            /* Close all opened handle and remove all */
+            if (_nodeHandleList != null)
```

Comment:
node-handle is managed in pipeline handle.
When destroying the pipeline (Interop.Pipeline.Destroy()), it will close all nodes, so this block is unnecessary.